macros/sql/surrogate_key.sql

```diff
@@ -1,40 +1,23 @@
 {%- macro surrogate_key(field_list) -%}
-    {# needed for safe_add to allow for non-keyword arguments see SO post #}
-    {# https://stackoverflow.com/questions/13944751/args-kwargs-in-jinja2-macros #}
-    {% set frustrating_jinja_feature = varargs %}
-    {{ return(adapter.dispatch('surrogate_key', 'dbt_utils')(field_list, *varargs)) }}
+    {{ return(adapter.dispatch('surrogate_key', 'dbt_utils')(field_list)) }}
 {% endmacro %}
 
 {%- macro default__surrogate_key(field_list) -%}
 
-{%- if varargs|length >= 1 or field_list is string %}
+{%- if field_list is not iterable or field_list is string or field_list is mapping -%}
 
 {%- set error_message = '
 Warning: the `surrogate_key` macro now takes a single list argument instead of \
-multiple string arguments. Support for multiple string arguments will be \
-deprecated in a future release of dbt-utils. The {}.{} model triggered this warning. \
+string arguments. The {}.{} model triggered this warning. \
 '.format(model.package_name, model.name) -%}
 
 {%- do exceptions.warn(error_message) -%}
 
-{# first argument is not included in varargs, so add first element to field_list_xf #}
-{%- set field_list_xf = [field_list] -%}
-
-{%- for field in varargs %}
-{%- set _ = field_list_xf.append(field) -%}
-{%- endfor -%}
-
-{%- else -%}
-
-{# if using list, just set field_list_xf as field_list #}
-{%- set field_list_xf = field_list -%}
-
 {%- endif -%}
 
-
 {%- set fields = [] -%}
 
-{%- for field in field_list_xf -%}
+{%- for field in field_list -%}
 
     {%- set _ = fields.append(
```

Comment:
```suggestion
    {%- do fields.append(
```

This `set _ = ` pattern is actually a hangover - we can do do `{% do fields.append('abc') %} and it will work without creating a redundant temporary variable. 

This also applies to L44, but that's not part of the diff so I can't comment on it.